File: 8_kyu/Pillars.cs
using System;

public static class Kata
{
    // Calculates distance between first and last pillar.
    // Distance is measured between inner edges of first and last pillar.
    // num_of_pillars - Number of pillars
    // distance - Distance/m between to pillars
    // width - Width/cm of one pillar
    // Return - Total inner distance/cm
    public static int Pillars(int numPill, int dist, int width) =>
      Math.Max(0, numPill-1)*100*dist + Math.Max(0, numPill-2)*width;
}

File: 8_kyu/Pillars.rs
use std::cmp::max;

/// Calculates the distance between pillars.
///
/// `distance` between pillars in meters.
/// `width` of the pillar in centimeters.
/// Returns distance in centimeters.
/// Returns the distance between first and last pillar (without widths of this twos).
fn pillars(num_of_pillars: u32, distance: u32, width: u32) -> u32 {
    (max(1,num_of_pillars)-1)*distance*100 + (max(2,num_of_pillars)-2)*width
}